.github/workflows/main.yml

```diff
@@ -37,20 +37,17 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - run: |
-        # Install fftw
-        case $(uname) in
-          Linux)
-          sudo apt update && sudo apt install -y -q fftw-dev
-            ;;
-          Darwin)
-            brew install fftw
-            ;;
-        esac
+    - id: setup-fftw
+      uses: ./.github/actions/setup-fftw
```

Comment:
Do we really need an internal action for this? 

Wouldn't be enough to just do the install in a single step?


```yaml
- id: setup-fftw
  run: |
    case $(uname) in
    Linux)
      sudo apt update
      sudo apt install -y -q fftw-dev
      echo FFTW_INCLUDE_DIR=/usr/include              >> $GITHUB_ENV
      echo FFTW_LIBRARY_DIR=/usr/lib/x86_64-linux-gnu >> $GITHUB_ENV
      ;;
    Darwin)
      brew install fftw
      local prefix=$(brew --prefix fftw)
      echo FFTW_INCLUDE_DIR=$prefix/include  >> $GITHUB_ENV
      echo FFTW_LIBRARY_DIR=$prefix/lib      >> $GITHUB_ENV
      ;;
    esac
```

Reply:
> In my opinion, it's cleaner to defer to the OS selection above than to ask for it inside the `run` command.

I disagree. If you are running the script under macOS, you can already query and know that fact (using `uname`), and there is no need to pass that information as an argument, otherwise you are somehow repeating yourself. On the other side, what would be the purpose of running on macOS and passing `ubuntu` as an argument? In other words, I do not see the point of this argument. 

> I also think it'd be great to have a reusable FFTW installation action for use in GitHub. (I couldn't find one.)
> 

Well, that's a separate concern. 

A proper action with the intention to be reused elsewhere should live in its own repository, it would require maintenance work (probably minimal), some minor documentation in a README, tagging and release in GitHub, and so on and so forth. And I'm not sure I want to be involved in yet another public repository to be responsible for maintenance.

On the specifics of this `mpi4py-fft` repository, I don't see the need of a reusable action, simply because we are not reusing it at all, we use it only once here! I would very much prefer to install fftw in a single step within the workflow as I discussed above.

> But no, we don't need it, and I'm happy to change it back to this inline `run` if you prefer this. Please let me know.

Yes, please, change it to a single step. 

PS: Note that my rejection is not because your original idea proposal was bad in any way. It is simply that at this point of my life I'm very very careful about adding maintenance responsibilities to my queue. Moreover, if someone else would have provided a public `setup-fftw` action, I would most likely use it straight away (if it comes from a reputable source). But if it is me the one that would have to be on charge of it, then no, thank you. The recent `xz` hack/fiasco showed us all that accepting and trusting external contributions and contributors requires constant vigilance. Therefore, once you accept open-source maintenance responsibilities ouside a company umbrella, then you are screwed for life, at the risk of your reputation, and all for free as in beer.

Reply:
No worries, @dalcinl.

Working on getting the pipelines working again with the switch statement.

Reply:
After testing in a Docker container, I figured out that the problem was the package name, it has to be `libfftw3-dev` in Unbuntu.

Reply:
Yep. I found the same.